src/python/WMCore/BossAir/StatusPoller.py

```diff
@@ -116,14 +118,18 @@ def checkStatus(self):
                     # Timeout status is used by JobTracker to fail jobs in WMBS database
                     logging.info("Killing job %i because it has exceeded timeout for status '%s'", job['id'], globalState)
                     job['status'] = 'Timeout'
-                    jobsToKill.append(job)
-
+                    jobsToKill[globalState].append(job)
+        
+        timeOutCodeMap = {"Running": 71304, "Pending": 71305, "Error": 71306}            
         # We need to show that the jobs are in state timeout
         # and then kill them.
+        jobsToKillList = itertools.chain.from_iterable(jobsToKill.values())
```

Comment:
This would break in BossAirAPI because it's an iterable and you cannot get its length. Perhaps you could use directly this util https://github.com/dmwm/WMCore/blob/master/src/python/Utils/IterTools.py#L22 (or just cast it here to list :))